Source/NSDockTile.m

```diff
@@ -109,4 +132,74 @@ - (void) display
     }
 }
 
+- (void)draw
+{
+  NSMutableDictionary *attrs;
+  NSPoint text_location;
+  NSRect disc_rect;
+  NSSize disc_size;
+  int image_width, pad;
+  NSBezierPath *p;
+  NSPoint point;
+  CGFloat radius;
+  NSSize imageSize;
+
+  if (_showsApplicationBadge)
+    {
+      [_appIconImage compositeToPoint:NSZeroPoint operation:NSCompositeCopy];
+
+      imageSize = [_appIconImage size];
+      attrs = [[NSMutableDictionary alloc] init];
+      [attrs setObject: [NSFont boldSystemFontOfSize: imageSize.width/5]  forKey: NSFontAttributeName];
+      [attrs setObject: [NSColor blackColor]  forKey: NSForegroundColorAttributeName];
+
+      disc_size = [_badgeLabel sizeWithAttributes: attrs];
+
+      image_width = imageSize.width;
+
+      pad = image_width / 10;
+      disc_size.height += pad;
+      disc_size.width += pad;
+
+
+      disc_rect = NSMakeRect(image_width-disc_size.width-4,
+			     image_width-disc_size.height-4,
+			     disc_size.width,
+			     disc_size.height);
+
+      text_location = NSMakePoint(image_width - (disc_size.width - ((disc_size.width - [_badgeLabel sizeWithAttributes: attrs].width)  * 0.5))-4,
```

Comment:
Why `[_badgeLabel sizeWithAttributes: attrs]`and not `disc_size`? That would help to let you see how to further simplify this code :-)

Reply:
well disc_size has some padding added. However I think it can be simplified after subsitution to just pad.

```
      text_location = NSMakePoint(image_width - (disc_size.width - pad  * 0.5)) - 4,
				  image_width - (disc_size.height - pad * 0.5)) - 4);
```

Reply:
Quite tidied up... please check.

Reply:
It could still be simplified, but most likely the compiler will do that by itself.